src/main/java/com/neonlab/product/service/ProductService.java

```diff
@@ -0,0 +1,67 @@
+package com.neonlab.product.service;
+import com.neonlab.common.annotations.Loggable;
+import com.neonlab.common.dto.ApiOutput;
+import com.neonlab.common.expectations.ProductNotFoundException;
+import com.neonlab.common.expectations.ProductUniqueCodeAlreadyExistsException;
+import com.neonlab.product.dtos.ProductDto;
+import com.neonlab.product.dtos.ProductRequestDto;
+import com.neonlab.product.entities.Product;
+import com.neonlab.product.enums.Units;
+import com.neonlab.product.repository.ProductRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.Optional;
+
+
+@Service
+@Loggable
+public class ProductService {
+
+    @Autowired
+    private ProductRepository productRepository;
+
+    public ProductDto addProduct(ProductRequestDto productReqDto) throws ProductUniqueCodeAlreadyExistsException {
+        Boolean isExist = productRepository.existsByCode(productReqDto.getCode());
+
+        if (isExist) {
+            throw new ProductUniqueCodeAlreadyExistsException("Product unique code already exists");
+        }
+        try {
+            var product = new Product();
+            product.setName(productReqDto.getName());
+            product.setUnits(Units.fromString(productReqDto.getUnits().unit));
+            product.setCode(productReqDto.getCode());
+            product.setBrand(productReqDto.getBrand());
+            product.setCategory(productReqDto.getCategory());
+            product.setSubCategory(productReqDto.getSubCategory());
+            product.setDescription(productReqDto.getDescription());
+            product.setPrice(productReqDto.getPrice());
+            product.setDiscountPrice(productReqDto.getDiscountPrice());
+            product.setQuantity(productReqDto.getQuantity());
+            product.setTags(productReqDto.getTags());
+            product.setVariety(productReqDto.getVariety());
+            productRepository.save(product);
+            return ProductDto.parse(product);
+
+        }catch (Exception e) {
+            // Throw an internal server error exception
+            throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, "Internal Server Error. Unable to add product.");
```

Comment:
Please use `ServerException.class` . This is in common repo. Please have a look and use the same. This is used at the same has been handled in `GlobalExceptionHandler`.

cc- @sjha24